src/main/java/percy/ui/Ui.java

```diff
@@ -0,0 +1,185 @@
+
+    /**
+     * Constructs an user interface.
+     * @param in input stream.
+     * @param out print stream.
+     */
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    /**
+     * Reads the command input by user.
+     * @return the command given.
+     */
+    public static String readCommand() {
+        String command = "";
+        while (command.trim().isEmpty()) {
+            command = in.nextLine();
+        }
+        return command;
+    }
+
+    /**
+     * Prints the start up message.
+     */
+    public static void printStartUpMsg() {
+        System.out.println(MESSAGE_LOGO + "\n" + DIVIDER
+                + MESSAGE_WELCOME + "\n"
+                + DIVIDER);
+    }
+
+    /**
+     * Makes the startup message.
+     * @return  formatted startup string.
+     */
+    public static String makeStartUpMsg() {
+        return MESSAGE_LOGO + "\n" + DIVIDER
+                + MESSAGE_WELCOME + "\n"
+                + DIVIDER;
+    }
+
+    /**
+     * Makes the goodbye message.
+     * @return formatted goodbye string.
+     */
+    public static String makeByeMsg() {
+        return DIVIDER + INDENT + MESSAGE_GOODBYE + "\n" + DIVIDER;
+    }
+
+    /**
+     * Formats any string for output purposes.
+     * @return formatted string.
+     */
+    public static String makeMsg(String s) {
+        return DIVIDER + INDENT + s.toString() + "\n" + DIVIDER;
+    }
+
+    /**
+     * Formats any array of string for multi-line output purposes.
+     * @param stringArr array of strings.
+     * @return formatted multi-line string.
+     */
+    public static String makeMsg(ArrayList<String> stringArr) {
+        String str = DIVIDER;
+        for (String s : stringArr) {
+            str += INDENT + s.toString() + "\n";
+        }
+        str += DIVIDER;
+        return str;
+    }
+
+    /**
+     * Formats the output message for listing of tasks.
+     * @param list latest task list.
+     * @return the output message for list command
+     */
+    public static String makeListMsg(TaskList list) {
+        String response;
+        response = DIVIDER + INDENT + "Here are the tasks in your list:\n";
+        int i = 1;
+        for (Task t : list.getTaskList()) {
+            response += INDENT
+                    + String.valueOf(i++) + ". "
+                    + t.toString() + "\n";
+        }
+        String taskString = (list.getTaskList().size() == 1) ? "task" : "tasks";
+        response += INDENT + "Now you have " + list.getTaskList().size() + " "
+                + taskString + " in the list.\n" + DIVIDER;
+        return response;
+    }
+
+
+    /**
+     * Make output message for added any task.
+     * @param t task added.
+     * @param list the updated list after adding task.
+     * @return message for adding task
```

Comment:
Should there be a full stop at the end of this line?
I noticed there are several instances of this issue.